app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/util/VaultItemListingDataExtensions.kt

```diff
@@ -337,7 +337,8 @@ private fun CipherView.toDisplayItem(
         optionsTestTag = "CipherOptionsButton",
         isAutofill = isAutofill,
         isFido2Creation = isFido2Creation,
-        shouldShowMasterPasswordReprompt = reprompt == CipherRepromptType.PASSWORD,
+        shouldShowMasterPasswordReprompt = (reprompt == CipherRepromptType.PASSWORD) &&
+            hasMasterPassword,
```

Comment:
👍

Reply:
Can we add a unit test for this scenario?

Reply:
done!